jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/HttpStream.java

```diff
@@ -82,6 +82,19 @@ public interface HttpStream extends Callback
      */
     void send(MetaData.Request request, MetaData.Response response, boolean last, ByteBuffer content, Callback callback);
 
+    /**
+     * Cancel any {@link #send(MetaData.Request, MetaData.Response, boolean, ByteBuffer, Callback)} call in progress.
+     *
+     * @param cause The cause of the cancellation
+     * @param callback The callback to ultimately {@link Callback#failed(Throwable) fail} after the cancellation
+     * @return A {@link Callback} that must be {@link Callback#failed(Throwable) fail} to complete the cancellation
+     *         and ultimately invoke the passed {@link Callback} (potentially in a different Thread).
+     */
+    default Callback cancelSend(Throwable cause, Callback callback)
```

Comment:
Another default implementation that does not respect the javadoc'ed contract?

Reply:
This does respect the contract. Calling failed on the returned callback completes the cancellation (nothing to be done) and ultimately invokes the passed callback.